# Quarterly Sales Tax Revenue Analysis

### [Data Source](https://tax.utah.gov/econstats/sales)

### Q2: April, May, June

In [1]:
import pandas as pd
import os
import numpy as np
import arcpy
from arcgis.features import GeoAccessor, GeoSeriesAccessor
arcpy.env.overwriteOutput = True

In [2]:
# show all columns
pd.options.display.max_columns = None

In [3]:
# Removes misc characters from string
def replace(string):
    string = string.replace("&", "")
    string = string.replace("-", " ")
    string = string.replace("(", "")
    string = string.replace(")", "")
    string = string.replace("/", " ")
    string = string.replace(",", "")
    string = string.replace(".", "")
    string = string.replace(" ", "_")
    return string


### Load Q2 2020 sales tax revenue data

In [4]:
# load excel sheet into a dataframe
xlsx = '.\\Inputs\\2020-02-quarterly-sales.xlsx'
q2_2020 = pd.read_excel(xlsx, sheet_name='Table 9', header=5)

# Drop last empty two rows
q2_2020.drop(q2_2020.tail(2).index,inplace=True)

# forward fill values from merged cells
q2_2020['County'] = q2_2020['County'].fillna(method='ffill')
q2_2020['Location Code'] = q2_2020['Location Code'].fillna(method='ffill')
q2_2020['City'] = q2_2020['City'].fillna(method='ffill')

# Figures with less than 10 taxpayers have been rounded up per Tax Commission disclosure rules.
# Rename column and convert values to boolean
q2_2020.rename(columns={'Unnamed: 5':'Rounded Up'}, inplace=True)
crosswalk = {'*':True, np.nan:False} # County_ID
q2_2020['Rounded Up'] = q2_2020['Rounded Up'].map(crosswalk)

q2_2020.head()


County  Location Code         City  \
0  Beaver County         1002.0  Beaver City   
1  Beaver County         1002.0  Beaver City   
2  Beaver County         1002.0  Beaver City   
3  Beaver County         1002.0  Beaver City   
4  Beaver County         1002.0  Beaver City   

                    Economic Sector (NAICS Code)     2020Q2  Rounded Up  
0  AGRICULTURE, FORESTRY, FISHING & HUNTING (11)   400000.0        True  
1                                 UTILITIES (22)   600000.0        True  
2                              CONSTRUCTION (23)    29553.0       False  
3                          MANUFACTURING (31-33)  1308568.0       False  
4            WHOLESALE TRADE-DURABLE GOODS (423)   413988.0       False

In [5]:
# pivot the table to get sectors as columns with tax values by city
sectors_q2_2020 = pd.pivot_table(q2_2020,values = '2020Q2',index ='City', columns = 'Economic Sector (NAICS Code)', aggfunc='first')

# fill NAs with 0
sectors_q2_2020 = sectors_q2_2020.fillna(0)

In [7]:
sectors_q2_2020.columns

Index(['ACCOMMODATION (721)',
       'ADMIN. & SUPPORT & WASTE MANAG. & REMED. SERVICES (56)',
       'AGRICULTURE, FORESTRY, FISHING & HUNTING (11)',
       'ARTS, ENTERTAINMENT AND RECREATION (71)', 'CONSTRUCTION (23)',
       'EDUCATIONAL SERVICES (61)', 'FINANCE & INSURANCE (52)',
       'FOOD SERVICES & DRINKING PLACES (722)',
       'HEALTH CARE & SOCIAL ASSISTANCE (62)', 'INFORMATION (51)',
       'MANAGEMENT OF COMPANIES & ENTERPRISES (55)', 'MANUFACTURING (31-33)',
       'MINING, QUARRYING, & OIL & GAS EXTRACTION (21)',
       'OTHER SERVICES-EXCEPT PUBLIC ADMINISTRATION (81)',
       'PRIOR-PERIOD PAYMENTS & REFUNDS', 'PRIVATE MOTOR VEHICLE SALES',
       'PROFESSIONAL, SCIENTIFIC & TECHNICAL SERVICES (54)',
       'PUBLIC ADMINISTRATION (92)', 'REAL ESTATE, RENTAL & LEASING (53)',
       'RETAIL-BUILD. MATERIAL, GARDEN EQUIP. & SUPPLIES DEALERS (444)',
       'RETAIL-CLOTHING & CLOTHING ACCESSORIES STORES (448)',
       'RETAIL-ELECTRONICS & APPLIANCE STORES (443)',
       

In [5]:
# fix column names
#new_names = [replace(item) for item in list(sectors_q2_2020.columns)]
new_names = ['L_ACCOMMODATION_20', 'S_ADMIN_SUPPORT_20','A_AG_WILDLIFE_20','L_CULTURAL_REC_20','C_CONSTRUCTION_20',
'E_EDUCATION_20','S_FINANCIAL_20','L_RSTRNT_BAR_20','H_HEALTH_CARE_20','S_IT_DATA_20','S_CORPORATE_MGMT_20',
'M_MANUFACTURING_20','E_EXTRACTION_20','O_OTHER_20','X_ADJUSTMENTS_20','R_AUTO_PRIVATE_20','S_PROF_TECH_SRV_20',
'G_GOVERNMENT_20','S_RENTAL_LEASING_20','R_BUILDING_SUPPLY_20','R_CLOTHING_20','R_ELECTRONICS_20','R_GROCERY_BEV_20',
'R_FURNITURE_20','R_GAS_STATIONS_20','R_GENERAL_RETAIL_20','R_HEALTH_RETAIL_20','R_OTHER_RETAIL_20','R_AUTO_RETAIL_20',
'R_NONSTORE_RETAIL_20','R_SPORT_HOBBY_20','L_SPECIAL_EVENT_20','W_DISTRIBUTION_20','X_UNKNOWN_20','U_UTILITIES_20',
'W_WHLSALE_DURABLE_20','S_WHLSLE_ETRADE_20','W_WHLSLE_NDURABLE_20']
sectors_q2_2020.columns = new_names

# get the total sales tax revenue
sectors_q2_2020['TOTAL_20'] = sectors_q2_2020[new_names].sum(axis=1)

sectors_q2_2020.head()

L_ACCOMMODATION_20  S_ADMIN_SUPPORT_20  A_AG_WILDLIFE_20  \
City                                                                      
American Fork            360525.0            625281.0         1750000.0   
Beaver City              988520.0             17144.0          400000.0   
Blanding                 408578.0             45672.0            1000.0   
Bountiful                100000.0            975548.0            4000.0   
Brigham                  655681.0            186299.0            3000.0   

               L_CULTURAL_REC_20  C_CONSTRUCTION_20  E_EDUCATION_20  \
City                                                                  
American Fork          1132810.0         10855679.0         69947.0   
Beaver City               7000.0            29553.0         10000.0   
Blanding                 20000.0           305743.0         50000.0   
Bountiful               585659.0          1647021.0         81571.0   
Brigham                 600000.0          1456710.0         28719.0   

               S_FINANCIAL_20  L_RSTRNT_BAR_20  H_HEALTH_CARE_20  \
City                                                               
American Fork       1510886.0       25740513.0          368700.0   
Beaver City          457083.0        2263825.0           60000.0   
Blanding              17919.0         600000.0           60000.0   
Bountiful           1116141.0       11839024.0          935892.0   
Brigham              201424.0        6912003.0          297620.0   

               S_IT_DATA_20  S_CORPORATE_MGMT_20  M_MANUFACTURING_20  \
City                                                                   
American Fork     6514315.0              25000.0           3709354.0   
Beaver City        495267.0               1000.0           1308568.0   
Blanding           485187.0               5000.0             77542.0   
Bountiful         5678745.0              25000.0           2171121.0   
Brigham           2767801.0              15000.0           9242535.0   

               E_EXTRACTION_20  O_OTHER_20  X_ADJUSTMENTS_20  \
City                                                           
American Fork          60000.0  12382993.0         2622074.0   
Beaver City                0.0   1021247.0          142192.0   
Blanding                   0.0    363436.0           59705.0   
Bountiful               5000.0   5958817.0         1671381.0   
Brigham                 1000.0   3281464.0          715733.0   

               R_AUTO_PRIVATE_20  S_PROF_TECH_SRV_20  G_GOVERNMENT_20  \
City                                                                    
American Fork          4500000.0           4496886.0          90000.0   
Beaver City             700000.0            153031.0              0.0   
Blanding                900000.0            143740.0         500000.0   
Bountiful              5500000.0           3301438.0         900000.0   
Brigham                3250000.0            796656.0           1000.0   

               S_RENTAL_LEASING_20  R_BUILDING_SUPPLY_20  R_CLOTHING_20  \
City                                                                      
American Fork            3964900.0            29535937.0      5219505.0   
Beaver City               143519.0             1005793.0       102242.0   
Blanding                   59894.0             2720256.0        80220.0   
Bountiful                2053241.0             1898525.0      2654922.0   
Brigham                  1868664.0             2727400.0       440580.0   

               R_ELECTRONICS_20  R_GROCERY_BEV_20  R_FURNITURE_20  \
City                                                                
American Fork         3463441.0         4360567.0       2896641.0   
Beaver City             26765.0         2250000.0          9064.0   
Blanding               111496.0         2250000.0          3258.0   
Bountiful             3819206.0        24807045.0       1691416.0   
Brigham               1228067.0        11722561.0        507641.0   

               R_GAS_STATIONS_20  R_GE

### Load Q2 2019 sales tax revenue data

In [6]:
# load excel sheet into a dataframe
xlsx = '.\\Inputs\\2019-02-quarterly-sales.xlsx'
q2_2019 = pd.read_excel(xlsx, sheet_name='Table 9', header=5)

# Drop last empty two rows
q2_2019.drop(q2_2019.tail(2).index,inplace=True)

# forward fill values from merged cells
q2_2019['County'] = q2_2019['County'].fillna(method='ffill')
q2_2019['Location Code'] = q2_2019['Location Code'].fillna(method='ffill')
q2_2019['City'] = q2_2019['City'].fillna(method='ffill')

# Figures with less than 10 taxpayers have been rounded up per Tax Commission disclosure rules.
# Rename column and convert values to boolean
q2_2019.rename(columns={'Unnamed: 5':'Rounded Up'}, inplace=True)
crosswalk = {'*':True, np.nan:False} # County_ID
q2_2019['Rounded Up'] = q2_2019['Rounded Up'].map(crosswalk)

q2_2019.tail()

County  Location Code         City  \
2411  Weber County        29040.0  South Ogden   
2412  Weber County        29040.0  South Ogden   
2413  Weber County        29040.0  South Ogden   
2414  Weber County        29040.0  South Ogden   
2415  Weber County        29040.0  South Ogden   

                          Economic Sector (NAICS Code)      2019Q2  Rounded Up  
2411             FOOD SERVICES & DRINKING PLACES (722)  10298427.0       False  
2412  OTHER SERVICES-EXCEPT PUBLIC ADMINISTRATION (81)   1429711.0       False  
2413                       PRIVATE MOTOR VEHICLE SALES   1500000.0        True  
2414                               SPECIAL EVENT SALES     45000.0        True  
2415                   PRIOR-PERIOD PAYMENTS & REFUNDS    165743.0       False

In [7]:
# pivot the table to get sectors as columns with tax values by city
sectors_q2_2019 = pd.pivot_table(q2_2019,values = '2019Q2',index ='City', columns = 'Economic Sector (NAICS Code)', aggfunc='first')

# fill NAs with 0
sectors_q2_2019 = sectors_q2_2019.fillna(0)

# fix column names
#new_names = [replace(item) for item in list(sectors_q2_2019.columns)]
new_names = ['L_ACCOMMODATION_19', 'S_ADMIN_SUPPORT_19','A_AG_WILDLIFE_19','L_CULTURAL_REC_19','C_CONSTRUCTION_19',
'E_EDUCATION_19','S_FINANCIAL_19','L_RSTRNT_BAR_19','H_HEALTH_CARE_19','S_IT_DATA_19','S_CORPORATE_MGMT_19',
'M_MANUFACTURING_19','E_EXTRACTION_19','O_OTHER_19','X_ADJUSTMENTS_19','R_AUTO_PRIVATE_19','S_PROF_TECH_SRV_19',
'G_GOVERNMENT_19','S_RENTAL_LEASING_19','R_BUILDING_SUPPLY_19','R_CLOTHING_19','R_ELECTRONICS_19','R_GROCERY_BEV_19',
'R_FURNITURE_19','R_GAS_STATIONS_19','R_GENERAL_RETAIL_19','R_HEALTH_RETAIL_19','R_OTHER_RETAIL_19','R_AUTO_RETAIL_19',
'R_NONSTORE_RETAIL_19','R_SPORT_HOBBY_19','L_SPECIAL_EVENT_19','W_DISTRIBUTION_19','X_UNKNOWN_19','U_UTILITIES_19',
'W_WHLSALE_DURABLE_19','S_WHLSLE_ETRADE_19','W_WHLSLE_NDURABLE_19']
sectors_q2_2019.columns = new_names

# get the total sales tax revenue
sectors_q2_2019['TOTAL_19'] = sectors_q2_2019[new_names].sum(axis=1)

sectors_q2_2019.head()

L_ACCOMMODATION_19  S_ADMIN_SUPPORT_19  A_AG_WILDLIFE_19  \
City                                                                      
American Fork            700000.0            566647.0         1250000.0   
Beaver City             1423694.0             31711.0          250000.0   
Blanding                1386710.0             48945.0            1000.0   
Bountiful                150000.0           2460364.0            2000.0   
Brigham                 2250000.0            188781.0            3000.0   

               L_CULTURAL_REC_19  C_CONSTRUCTION_19  E_EDUCATION_19  \
City                                                                  
American Fork          1600920.0          8794650.0         99011.0   
Beaver City               7000.0            35000.0         15000.0   
Blanding                 90000.0           210655.0        250000.0   
Bountiful              1105872.0          2074898.0        104656.0   
Brigham                 800000.0          1527976.0         58893.0   

               S_FINANCIAL_19  L_RSTRNT_BAR_19  H_HEALTH_CARE_19  \
City                                                               
American Fork       1790289.0       24651818.0          498511.0   
Beaver City          220378.0        2737418.0           50000.0   
Blanding              25496.0         700000.0           60000.0   
Bountiful           1151100.0       13673812.0          987748.0   
Brigham              292577.0        7047410.0          499834.0   

               S_IT_DATA_19  S_CORPORATE_MGMT_19  M_MANUFACTURING_19  \
City                                                                   
American Fork     9244154.0              20000.0           3599673.0   
Beaver City        722352.0               1000.0            698857.0   
Blanding           483587.0               5000.0            164496.0   
Bountiful         5176705.0              20000.0           2685170.0   
Brigham           2584387.0              25000.0           6400432.0   

               E_EXTRACTION_19  O_OTHER_19  X_ADJUSTMENTS_19  \
City                                                           
American Fork          25000.0  11024408.0         3859810.0   
Beaver City                0.0    889776.0          719732.0   
Blanding                   0.0    260314.0          189826.0   
Bountiful              10000.0   6990239.0         4448948.0   
Brigham                 1000.0   3013123.0          678568.0   

               R_AUTO_PRIVATE_19  S_PROF_TECH_SRV_19  G_GOVERNMENT_19  \
City                                                                    
American Fork          3000000.0           2182874.0          35000.0   
Beaver City             500000.0             96364.0              0.0   
Blanding                800000.0             61873.0         500000.0   
Bountiful              5250000.0           2157862.0         600000.0   
Brigham                2500000.0            391323.0           1000.0   

               S_RENTAL_LEASING_19  R_BUILDING_SUPPLY_19  R_CLOTHING_19  \
City                                                                      
American Fork            3830247.0            23085674.0      7224842.0   
Beaver City               100714.0              564664.0        58911.0   
Blanding                   93226.0             1948647.0        62024.0   
Bountiful                2670871.0             1318455.0      2791903.0   
Brigham                  1731936.0             1916996.0       392600.0   

               R_ELECTRONICS_19  R_GROCERY_BEV_19  R_FURNITURE_19  \
City                                                                
American Fork         7042705.0         4026871.0       3546430.0   
Beaver City             23509.0         1750000.0          4000.0   
Blanding                39162.0         2000000.0          7000.0   
Bountiful             3289392.0        20779545.0       1808556.0   
Brigham               1059660.0         9955541.0        529234.0   

               R_GAS_STATIONS_19  R_GE

### Calculate the differences 
*2020 - 2019 = Diff*

In [8]:
# merge the 2019-2020 difference with the sdf
sectors_q2_diff = pd.DataFrame(sectors_q2_2020.values - sectors_q2_2019.values)

# add back the column names
new_names = ['L_ACCOMMODATION_D20', 'S_ADMIN_SUPPORT_D20','A_AG_WILDLIFE_D20','L_CULTURAL_REC_D20','C_CONSTRUCTION_D20',
'E_EDUCATION_D20','S_FINANCIAL_D20','L_RSTRNT_BAR_D20','H_HEALTH_CARE_D20','S_IT_DATA_D20','S_CORPORATE_MGMT_D20',
'M_MANUFACTURING_D20','E_EXTRACTION_D20','O_OTHER_D20','X_ADJUSTMENTS_D20','R_AUTO_PRIVATE_D20','S_PROF_TECH_SRV_D20',
'G_GOVERNMENT_D20','S_RENTAL_LEASING_D20','R_BUILDING_SUPPLY_D20','R_CLOTHING_D20','R_ELECTRONICS_D20','R_GROCERY_BEV_D20',
'R_FURNITURE_D20','R_GAS_STATIONS_D20','R_GENERAL_RETAIL_D20','R_HEALTH_RETAIL_D20','R_OTHER_RETAIL_D20','R_AUTO_RETAIL_D20',
'R_NONSTORE_RETAIL_D20','R_SPORT_HOBBY_D20','L_SPECIAL_EVENT_D20','W_DISTRIBUTION_D20','X_UNKNOWN_D20','U_UTILITIES_D20',
'W_WHLSALE_DURABLE_D20','S_WHLSLE_ETRADE_D20','W_WHLSLE_NDURABLE_D20', 'TOTAL_D20']

sectors_q2_diff.columns = new_names

# add back cities as index
sectors_q2_diff['City'] = sectors_q2_2020.index
sectors_q2_diff = sectors_q2_diff.set_index('City')

#check table
sectors_q2_diff.head(10)

L_ACCOMMODATION_D20  S_ADMIN_SUPPORT_D20  \
City                                                           
American Fork                 -339475.0              58634.0   
Beaver City                   -435174.0             -14567.0   
Blanding                      -978132.0              -3273.0   
Bountiful                      -50000.0           -1484816.0   
Brigham                      -1594319.0              -2482.0   
Cedar City                   -3002509.0             367130.0   
Centerville                     -5000.0              50243.0   
Clearfield                     -50000.0              96076.0   
Cottonwood Heights            -686505.0             -29231.0   
Delta                         -150000.0              -2349.0   

                    A_AG_WILDLIFE_D20  L_CULTURAL_REC_D20  C_CONSTRUCTION_D20  \
City                                                                            
American Fork                500000.0           -468110.0           2061029.0   
Beaver City                  150000.0                 0.0             -5447.0   
Blanding                          0.0            -70000.0             95088.0   
Bountiful                      2000.0           -520213.0           -427877.0   
Brigham                           0.0           -200000.0            -71266.0   
Cedar City                     8530.0           -186282.0            740470.0   
Centerville                       0.0            -10000.0           -484578.0   
Clearfield                     1000.0           -250000.0            163890.0   
Cottonwood Heights             3000.0            -40000.0            180454.0   
Delta                         70000.0            -70000.0           1811033.0   

                    E_EDUCATION_D20  S_FINANCIAL_D20  L_RSTRNT_BAR_D20  \
City                                                                     
American Fork              -29064.0        -279403.0         1088695.0   
Beaver City                 -5000.0         236705.0         -473593.0   
Blanding                  -200000.0          -7577.0         -100000.0   
Bountiful                  -23085.0         -34959.0        -1834788.0   
Brigham                    -30174.0         -91153.0         -135407.0   
Cedar City                -350973.0        -111789.0         -744532.0   
Centerville                  3773.0        -159829.0         -277358.0   
Clearfield                  -5104.0          79958.0         -361897.0   
Cottonwood Heights         -13645.0         514343.0        -4758782.0   
Delta                        1000.0         -15408.0          108830.0   

                    H_HEALTH_CARE_D20  S_IT_DATA_D20  S_CORPORATE_MGMT_D20  \
City                                                                         
American Fork               -129811.0     -2729839.0                5000.0   
Beaver City                   10000.0      -227085.0                   0.0   
Blanding                          0.0         1600.0                   0.0   
Bountiful                    -51856.0       502040.0                5000.0   
Brigham                     -202214.0       183414.0              -10000.0   
Cedar City                   106396.0        75992.0               -5000.0   
Centerville                  -53637.0     -2228983.0             -275000.0   
Clearfield                    16337.0       731313.0              400000.0   
Cottonwood Heights          -138454.0      1292468.0             -100000.0   
Delta                             0.0       202944.0               -5000.0   

                    M_MANUFACTURING_D20  E_EXTRACTION_D20  O_OTHER_D20  \
City                                                                     
American Fork                  109681.0           35000.0    1358585.0   
Beaver City                    609711.0               0.0     131471.0   
Blanding                       -86954.0               0.0     103122.0   
Bountiful                     -514049.0           -5000.0   -1031422.0   
Brigham 

In [9]:
# Check join output shape
print(sectors_q2_2019.shape)
print(sectors_q2_2020.shape)
print(sectors_q2_diff.shape)

(66, 39)
(66, 39)
(66, 39)


### Load municipalities and townships shapefile and join to formatted sales tax data

In [10]:
# load cities shapefile into pandas spatial dataframe
cities_shp = '.\\Inputs\\Cities.shp'
cities_sdf = pd.DataFrame.spatial.from_featureclass(cities_shp)

# Rename columns
cities_sdf.columns = ['FID', 'NAME', 'SOURCE', 'SALESTAXID', 'POPLASTCEN', 'POPLASTEST',
       'AREA_SQMI', 'SHAPE']

In [11]:
# merge the 2020 sdf with the pivoted data
cities_sdf2 = cities_sdf.merge(sectors_q2_2019, left_on= "NAME", right_on="City", how="inner")
cities_sdf2 = cities_sdf2.merge(sectors_q2_2020, left_on= "NAME", right_on="City", how="inner")
cities_sdf2 = cities_sdf2.merge(sectors_q2_diff, left_on= "NAME", right_on="City", how="inner")
cities_sdf2.tail(10)


FID              NAME  SOURCE SALESTAXID  POPLASTCEN  POPLASTEST  \
55  222          Syracuse  Cities        049       25113       30400   
56  224      Taylorsville  Cities        142       59989       60192   
57  225       Tooele City  Cities        048       32629       35251   
58  228         Tremonton  Cities        113        7828        8882   
59  232            Vernal  Cities        024        9460       10370   
60  238   Washington City  Cities        027       19640       27686   
61  243    West Bountiful  Cities        061        5420        5731   
62  245       West Jordan  Cities        155      105891      116046   
63  247  West Valley City  Cities        167      133796      136401   
64  252       Woods Cross  Cities        057       10086       11328   

    AREA_SQMI                                              SHAPE  \
55  10.193900  {'rings': [[[411406.60675760085, 4549994.30436...   
56  10.839700  {'rings': [[[422191.826826254, 4504303.3642982...   
57  24.087099  {'rings': [[[392501.67668449413, 4486111.93423...   
58   7.955100  {'rings': [[[399303.9767575987, 4621607.124490...   
59   4.632710  {'rings': [[[627188.8071560834, 4479658.624106...   
60  36.165600  {'rings': [[[278123.416362615, 4102773.6636791...   
61   3.299440  {'rings': [[[421904.10680578306, 4528452.80425...   
62  32.306999  {'rings': [[[415038.796736156, 4499131.1642154...   
63  35.799702  {'rings': [[[418708.9467166035, 4508796.864202...   
64   3.835550  {'rings': [[[424779.70679974474, 4524289.10421...   

    L_ACCOMMODATION_19  S_ADMIN_SUPPORT_19  A_AG_WILDLIFE_19  \
55             40000.0             91580.0            1000.0   
56            300000.0            511515.0           15000.0   
57           1500000.0            242794.0            4000.0   
58           1000000.0           1102050.0            5000.0   
59           2889773.0            230630.0            2000.0   
60           2417500.0            209375.0           60000.0   
61            700000.0             13588.0           80000.0   
62           2000000.0           1223348.0          350000.0   
63           7716953.0           2806839.0           30000.0   
64           1500000.0            324566.0            1000.0   

    L_CULTURAL_REC_19  C_CONSTRUCTION_19  E_EDUCATION_19  S_FINANCIAL_19  \
55          1250000.0           204550.0         20000.0        531340.0   
56          2107794.0          1455466.0        161570.0       1788978.0   
57          1250000.0           350260.0         43953.0        757914.0   
58           450000.0          1028023.0         15000.0        101604.0   
59           400000.0           530788.0         25328.0        387189.0   
60          1250000.0           844507.0          6000.0        425832.0   
61           350000.0            34639.0          4000.0        115324.0   
62          2332992.0         10001889.0        142172.0       2566994.0   
63          3463236.0         29939509.0        262545.0       3678411.0   
64             1000.0           730769.0         60000.0        227631.0   

    L_RSTRNT_BAR_19  H_HEALTH_CARE_19  S_IT_DATA_19  S_CORPORATE_MGMT_19  \
55        4888061.0           50000.0     2771091.0               8000.0   
56       28052721.0          105335.0     8700270.0              25000.0   
57       12489542.0          634389.0     4901416.0              15000.0   
58        3445450.0            2000.0      983060.0               2000.0   
59       10508521.0          329506.0     4877731.0               8000.0   
60        9247024.0           15000.0     2571075.0               9000.0   
61        4072332.0            6000.0     1578981.0               7000.0   
62       37479616.0         1019149.0    22649253.0             250000.0   
63       56377823.0         1676228.0    25717983.0              45000.0   
64        4771762.0           25000.0     2522966.0               5000.0   

    M_MANUFACTURING_19  E_EXTRACTION_19  O_OTHER_19  X_ADJUSTMENTS_19  \
55        

In [12]:
# Confirm join count, Salt Lake County (Unincorporated) should be the only one not joined
l1 = list(q2_2020['City'].value_counts().index)
l2 = list(cities_sdf['NAME'].value_counts().index)
list(set(l1) - set(l2))

['Salt Lake County (Unincorporated)']

In [13]:
# Create file gdb and export sdf to feature class
outputs = '.\\Outputs'

gdb = os.path.join(outputs, "taxable_sales.gdb")
if not arcpy.Exists(gdb):
    arcpy.CreateFileGDB_management(outputs, "taxable_sales.gdb")

cities_sdf2.spatial.to_featureclass(location=os.path.join(outputs, "taxable_sales.gdb","taxable_sales_utah_q2"))

'E:\\Projects\\UT_Sales_Tax_Analysis\\Outputs\\taxable_sales.gdb\\taxable_sales_utah_q2'

### Analysis

In [14]:
# Get year specific column names
standard_cols = ['fid','name','source','salestaxid','poplastcen','poplastcen','area_sqmi', 'SHAPE']
names_19 = standard_cols + [col for col in cities_sdf2.columns if '_19' in col]
names_20 = standard_cols + [col for col in cities_sdf2.columns if '_20' in col]
names_D20 = standard_cols + [col for col in cities_sdf2.columns if '_d20' in col]

# subset columns by time
cities_q2_19 = cities_sdf2[names_19]
cities_q2_20 = cities_sdf2[names_20]
cities_q2_d20 = cities_sdf2[names_D20]


In [15]:
# 10 cities with smallest difference of revenue
cities_q2_d20[['name', 'poplastcen', 'area_sqmi', 'total_d20']].sort_values('total_d20', ascending=False).head(10)

name  poplastcen  poplastcen  area_sqmi   total_d20
62       West Jordan      105891      105891  32.306999  91207296.0
54         St George       76548       76548  77.209900  84029566.0
51      Spanish Fork       36232       36232  16.249701  57141263.0
57       Tooele City       32629       32629  24.087099  48598585.0
63  West Valley City      133796      133796  35.799702  46907870.0
0      American Fork       27413       27413  11.256600  46408374.0
20            Layton       69457       69457  22.576200  42043505.0
5         Cedar City       30139       30139  36.053398  40919339.0
23             Logan       49612       49612  18.485800  40702065.0
34              Orem       92993       92993  18.628000  40443599.0

In [16]:
# 10 cities with largest difference of revenue
cities_q2_d20[['name', 'poplastcen', 'area_sqmi', 'total_d20']].sort_values('total_d20', ascending=True).head(10)

name  poplastcen  poplastcen   area_sqmi    total_d20
45  Salt Lake City      190749      190749  110.734001 -217551883.0
27            Moab        5124        5124    4.802460  -43345377.0
35       Park City        7709        7709   20.341700  -42810440.0
30          Murray       47311       47311   12.293700  -40848514.0
48    South Jordan       51057       51057   22.263201  -25277957.0
52      Springdale         569         569    4.594880  -21011489.0
12      Farmington       18752       18752   10.035100  -17939769.0
43       Roosevelt        6260        6260    6.757070   -9635486.0
17           Kanab        4413        4413   14.473200   -6206268.0
59          Vernal        9460        9460    4.632710   -3023499.0

#### Some industries of interest:
- RETAIL-GASOLINE STATIONS (447)
- RETAIL-FOOD & BEVERAGE STORES (445)
- REAL ESTATE, RENTAL & LEASING (53)
- ARTS, ENTERTAINMENT AND RECREATION (71)
- ACCOMMODATION (721)